SFS_general/delayline.m

```diff
@@ -55,92 +64,158 @@
+        % incorporating the delay d afterwards.
+        
+        % number of parallel filters, i.e. order of polynomial + 1
+        % Nfilter = fracdelay.pre.farrow.Npol+1;
+        
+        to_be_implemented(mfilename);
+    case 'none'
+        % Nothing to be done
+    otherwise
+        fprintf(['%s: \"%s\" is an unknown pre-processing method for delay', ...
+            'line'],upper(mfilename),fracdelay.pre.method);
+end
+
+%% ===== Fractional Delay ================================================
+
+dt = rfactor.*dt;  % resampled delays
+samples = rfactor.*samples;  % length of resampled signals
+
+if strcmp(fracdelay.pre.method,'farrow')
+    to_be_implemented(mfilename);
+else  % There is no post processing stage if the Farrow Structure used
+    % === Post Processing ================================================
+    a = ones(1,channels);  % denominator of fractional delay filter
+    switch fracdelay.filter
+        case 'zoh'
+            % === Zero-Order-Hold (Integer Delays) =======================
+            idt = ceil(dt);  % round up to next integer delay
+            b = ones(1,channels);
+        case 'lagrange'
+            % ==== Lagrange Polynomial Interpolator ======================
+            if mod(fracdelay.order,2) == 0
+                idt = round(dt);  % round delay for even order
+            else
+                idt = floor(dt);  % floor delay for odd order
+            end
+            fdt = dt - idt;  % fractional part of delays
+            b = lagrange_filter(fracdelay.order,fdt);
+        case 'thiran'
+            % ==== Thiran's Allpass Filter for Maximally Flat Group Delay
+            idt = round(dt);  % integer part of delays
+            fdt = dt - idt;  % fractional part of delays
+            [b, a] = thiran_filter(fracdelay.order,fdt);
+        case 'least_squares'
+            % ==== Least Squares Interpolation Filter ====================
+            idt = floor(dt);  % integer part of delays
+            fdt = dt - idt;  % fractional part of delays
+            b = zeros(fracdelay.order+1,channels);
+            for ii=1:channels
+                b(:,ii) = general_least_squares(fracdelay.order+1,fdt(ii),0.90);
+            end
+        otherwise
+            error('%s: \"%s\" is an unknown fractional delay filter', ...
+                upper(mfilename),fracdelay.filter);
+    end
+    
+    for ii=1:channels
+        sig(:,ii) = filter(b(:,ii),a(:,ii),sig(:,ii));
+    end
+end
+
+%% ===== Integer Delay ===================================================
+% Handling of too long delay values (returns vector of zeros)
+idt(abs(idt)>samples) = samples;
+
+% Handle positive or negative delays
+for ii=1:channels
+    if idt(ii)>=0
+        sig(:,ii) = [zeros(idt(ii),1); weight(ii)*sig(1:end-idt(ii),ii)];
+    else
+        sig(:,ii) = [weight(ii)*sig(-idt(ii)+1:end,ii); zeros(-idt(ii),1)];
+    end
+end
+sig = sig(1:rfactor:samples,:);
```

Comment:
I guess this line is needed if you use some kind of resampling above. But this just cut the signal. Before we did explicit resampling to the original fs:
```Matlab
sig2 = resample(sig,rfactor,1);
sig2 = delayline(sig2,rfactor.*dt,weight,conf2);
sig = resample(sig2,1,rfactor);
```
Can you really replace this by this line?